test/controllers/herbaria_controller_test.rb

```diff
@@ -405,11 +407,14 @@ def test_create
                  herbarium.mailing_address)
     assert_equal(create_params[:description].strip, herbarium.description)
     assert_empty(herbarium.curators)
-    email = ActionMailer::Base.deliveries.last
-    assert_equal(katrina.email, email.header["reply_to"].to_s)
-    assert_match(/new herbarium/i, email.header["subject"].to_s)
-    assert_includes(email.body.to_s, "Burbank Herbarium")
-    assert_includes(email.body.to_s, herbarium.show_url)
+    assert_equal(count_before + 1, QueuedEmail.count)
+    # FIXME: - can QueuedEmail content be tested in a controller test?
+    #         Or should it be elsewhere?
+    # email = ActionMailer::Base.deliveries.last
+    # assert_equal(katrina.email, email.header["reply_to"].to_s)
+    # assert_match(/new herbarium/i, email.header["subject"].to_s)
+    # assert_includes(email.body.to_s, "Burbank Herbarium")
+    # assert_includes(email.body.to_s, herbarium.show_url)
```

Comment:
@pellaea see note here. Seems like all i can test for here is that an email has been queued. No other controller test seems to test the content of the `QueuedEmail`, and only `ObservationsControllerTest` tests that an email has in fact been queued.

Reply:
Thanks. When i binding break here, whether I get the last QE or dump all QE's, all of them are Webmaster emails. I don't get why but at least it does queue another email... 

But i'd rather get it!

Between them, `ApplicationMailerTest` and `QueuedEmailTest` seem like they're covering the email creation, and as i mentioned no other controller test tests the attributes of a `QueuedEmail` in any way. But I agree, it seems like I should be able to get that last instance.

Reply:
Solved. 

The last  `QueuedEmail` _should_ of course be a `Webmaster` email, not an `AddRecordToHerbarium` email, and what we can check for is that the `sender` is set to `katrina`. Done.